apps/condo/domains/common/utils/serverSchema/perFieldAccess.js

```diff
@@ -0,0 +1,94 @@
+const { transformCRUDString } = require('@condo/domains/common/utils/serverSchema/accessUtils')
+const { GQL_LIST_SCHEMA_TYPE, GQL_CUSTOM_SCHEMA_TYPE, getSchemaCtx } = require('@core/keystone/schema')
+const { get } = require('lodash')
+
+function getFieldPermissions (user, listKey, field) {
+    return transformCRUDString(get(user, ['permissions', 'lists', listKey, 'fields', field], ''))
+}
+
+function getCustomPermissions (user, key) {
+    return get(user, ['permissions', 'custom', key], null)
+}
+
+function hasDefinedPermissions (user) {
+    return get(user, 'permissions') != null
+}
+
+function denyIfHasNoPermission (user, permission, originalAccessResult) {
+    if (!originalAccessResult) return false
+    if (hasDefinedPermissions(user)) {
+        return permission === true
+    }
+    return true
+}
+
+function perFieldAccess (schemaType, schemaName, schema) {
```

Comment:
It’s too complex with a lot of edge cases! I don’t think that it’s a good idea to add such complex solution inside the project